src/test/java/com/example/postgresdemo/service/QuestionServiceTest.java

```diff
@@ -0,0 +1,124 @@
+package com.example.postgresdemo.service;
+
+import com.example.postgresdemo.exception.ResourceNotFoundException;
+import com.example.postgresdemo.model.Question;
+import com.example.postgresdemo.model.QuestionRequestDTO;
+import com.example.postgresdemo.model.QuestionResponseDTO;
+import com.example.postgresdemo.repository.QuestionRepository;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.data.domain.Page;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+
+@ExtendWith(MockitoExtension.class)
+class QuestionServiceTest {
+
+    @Mock
+    QuestionRepository questionRepository;
+
+    @InjectMocks
+    QuestionService questionService;
+
+    @Test
+    void testFindAll() {
+        Question question1 = new Question();
+        question1.setId(1L);
+        question1.setTitle("Title1" );
+        question1.setDescription("Description1");
+        Question question2 = new Question();
+        question2.setId(2L);
+        question2.setTitle("Title2");
+        question2.setDescription("Description2");
+
+        Page<Question> page = new PageImpl<>(Arrays.asList(question1, question2));
+        Mockito.when(questionRepository.findAll(any(PageRequest.class))).thenReturn(page);
+
+        Page<QuestionResponseDTO> result = questionService.findAll(PageRequest.of(0, 10));
+
+        Mockito.verify(questionRepository).findAll(any(Pageable.class));
+        Assertions.assertEquals(2, result.getTotalElements());
+        Assertions.assertEquals("Title1\nDescription1", result.getContent().get(0).getBody());
+        Assertions.assertEquals("Title2\nDescription2", result.getContent().get(1).getBody());
+    }
+
+    @Test
+    void testCreate() {
+        QuestionRequestDTO request = new QuestionRequestDTO();
+        request.setTitle("Title");
+        request.setDescription("Description");
+        Question question = new Question();
+        question.setId(1L);
+        question.setTitle("Title");
+        question.setDescription("Description");
+        Mockito.when(questionRepository.save(any(Question.class))).thenReturn(question);
```

Comment:
Instead of `any(Question.class)` it would be better to use [ArgumentCaptor](https://www.baeldung.com/mockito-argumentcaptor). With the help of captor you will be able to get the real value passed to `questionRepository.save()` method and verify that all Question fields are mapped correctly from QuestionRequestDTO